File: Zadania/Lab01/Trojmian.java
import java.util.*;

public class Trojmian {
  	private double a, b, c, x1, x2, delta;

  	void readData() {
    	Scanner scanner = new Scanner(System.in);
    	System.out.println("Podaj współczynniki trójmianu.");
    	do {
      		System.out.print("a = ");
			while(!scanner.hasNextDouble()) {
				System.out.print(scanner.next());
				System.out.print(" nie jest poprawnym zapisem liczby.\na = ");
			}
			a = scanner.nextDouble();
			if (a == 0)
				System.out.print("a nie może być równe 0.\n");
		} while(a == 0);
    
    	System.out.print("b = ");
    	while(!scanner.hasNextDouble()) {
      		System.out.print(scanner.next());
      		System.out.print(" nie jest poprawnym zapisem liczby.\nb = ");
    	}
    	b = scanner.nextDouble();
    
    	System.out.print("c = ");
    	while(!scanner.hasNextDouble()) {
      		System.out.print(scanner.next());
      		System.out.print(" nie jest poprawnym zapisem liczby.\nc = ");
    	}
		c = scanner.nextDouble();
	}

	void solve() {
		delta = b * b - 4.0 * a * c;
		if (delta > 0.0) {
			x1 = (-b - Math.sqrt(delta))/(2.0 * a);
			x2 = (-b + Math.sqrt(delta))/(2.0 * a);
		} else if (delta == 0.0) {
			x1 = (-b)/(2.0 * a);
		}
	}
  
	void printResults() {
		if (delta > 0.0) 
			System.out.println("Rozwiązania rzeczywiste: x1 = " + x1 + ", x2 = " + x2 + "."); 
		else if (delta == 0.0)
			System.out.println("Rozwiązanie rzeczywiste: x1 = " + x1 + "."); 
		else
			System.out.println("Brak rozwiązań rzeczywistych.");
	}
  
	static public void main(String args[]) {
		Trojmian t = new Trojmian();
		t.readData();
		t.solve();
		t.printResults();  
	}
}

File: Zadania/Lab01/trojmian.c
#include <stdio.h>
#include <math.h>

int main() {
    float a, b, c, delta, x1, x2;

    printf("Podaj wspolczynniki trojmianu\n");

    do {
        printf("a = ");
        scanf("%f", &a);
        if (a == 0) {
            printf("Wspolczynnik a musi byc rozny od 0\n");
        }
    } while (a == 0);

    printf("b = ");
    scanf("%f", &b);
    printf("c = ");
    scanf("%f", &c);

    delta = b * b - 4.0 * a * c;
    if (delta > 0.0) {
        x1 = (-b - sqrt(delta))/(2.0 * a);
        x2 = (-b + sqrt(delta))/(2.0 * a);
        printf("Pierwiastkami sa liczby %.2f, %.2f\n", x1, x2);
    } else if (delta == 0.0) {
        x1 = (-b)/(2.0 * a);
        printf("Pierwiastkiem jest liczba %.2f\n", x1);
    } else {
        printf("Brak pierwiastkow\n");
    }
    
    return 0;
}
